java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -183,25 +173,17 @@ class ReferenceModeStore private constructor(
     override suspend fun idle() {
         backingStore.idle()
         containerStore.idle()
+        receiveQueue.flush().join()
```

Comment:
I'm not 100% sure about this approach, but being able to use the same `idle` method to also wait for pending cleanup to finish feels nice.